src/Microsoft.AspNetCore.CertificateGeneration.Task/Program.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.AspNetCore.CertificateGeneration.Task
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            app.HelpOption("-h|--help");
+
+            app.Command("ssl", c =>
```

Comment:
On the contrary, SSL is way more specific than self-sign or self-signed. self-sign or self-signed doesn't specify at all what the certificate is being used/intended for.
I can have a self-signed certificate that gets used for client authentication for example.
BTW, the consumer of this tool is the task below, so I don't think names matter that much in that regard.